guides/06_client-libraries/02_getting-started-with-the-js-client.md

```diff
@@ -0,0 +1,268 @@
+import { client } from "@gradio/client";
+
+const app = await client("abidlabs/en2fr");
+const result = await client.predict("/predict", ["Hello"]);
+```
+
+If there are multiple parameters, then you should pass them as an array to `.predict()`, like this:
+
+```js
+import { client } from "@gradio/client";
+
+const app = await client("gradio/calculator");
+const result = await app.predict("/predict", [4, "add", 5]);
+```
+
+For certain inputs, such as images, you should pass in a `Buffer`, `Blob` or `File` depending on what is most convenient. In node, this would be a `Buffer` or `Blob`; in a browser environment, this would be a `Blob` or `File`.
+
+
+```js
+import { client } from "@gradio/client";
+
+const response = await fetch(
+  "https://audio-samples.github.io/samples/mp3/blizzard_unconditional/sample-0.mp3"
+);
+const audio_file = await response.blob();
+
+const app = await client("abidlabs/whisper");
+const result = await client.predict("/predict", [ audio_file ]);
+```
+
+
+## Using events
+
+If the API you are working with can return results over time, or you wish to access information about the status of a job, you can use the event interface for more flexibility. This is especially useful for iterative endpoints or generator endpoints that will produce a series of values over time as discreet responses.
+
+```js
+import { client } from "@gradio/client";
+
+function log_result(payload) {
+  const {
+    data: [translation]
+  } = payload;
+
+  console.log(`The translated result is: ${translation}`);
+}
+
+const app = await client("abidlabs/en2fr");
+const job = app.submit("/predict", ["Hello"]);
+
+job.on("data", log_result);
+```
+
+## Status
+
+The event interface also allows you to get the status of the running job by listening to the `"status"` event. This returns an object with the following attributes: `status` (a human readbale status of the current job, `"pending" | "generating" | "complete" | "error"`), `code` (the detailed gradio code for the job), `position` (the current position of this job in the queue), `queue_size` (the total queue size),  `eta` (estimated time this job will complete), `success` (a boolean representing whether the job completed successfully), and `time` ( as `Date` object detailing the time that the status was generated). 
+
+
+```js
+import { client } from "@gradio/client";
+
+function log_status(status) {
+  console.log(`The current status for this job is: ${JSON.stringify(status, null, 2)}.`)
+}
+
+const app = await client("abidlabs/en2fr")
+const job = app.submit("/predict", ["Hello"]);
+
+job.on("status", log_status);
+```
+
+## Cancelling Jobs
+
+The job instance also has a `.cancel()` method that cancels jobs that have been queued but not started. For example, if you run:
+
+
+
+```js
+import { client } from "@gradio/client";
+
+const app = await client("abidlabs/en2fr")
+const job_one = app.submit("/predict", ["Hello"]);
+const job_two = app.submit("/predict", ["Friends"]);
+
+job_one.cancel();
```

Comment:
This is throwing a:
`ReferenceError: addEventListener is not defined` for me

Reply:
Ah nice catch, overlooked something here.